js/src/tab.js

```diff
@@ -17,158 +17,262 @@ import BaseComponent from './base-component'
+        element.classList.remove(CLASS_NAME_SHOW)
       }
 
-      if (active.getAttribute('role') === 'tab') {
-        active.setAttribute('aria-selected', false)
+      if (element.getAttribute('role') !== 'tab') {
+        return
       }
+
+      element.setAttribute('aria-selected', false)
+      element.setAttribute('tabindex', '-1')
+      this._toggleDropDown(element, false)
+      EventHandler.trigger(element, EVENT_HIDDEN, { relatedTarget: relatedElem })
     }
 
-    element.classList.add(CLASS_NAME_ACTIVE)
-    if (element.getAttribute('role') === 'tab') {
-      element.setAttribute('aria-selected', true)
+    this._queueCallback(complete, element, isAnimated)
+  }
+
+  _keydown(event) {
+    if (!([ARROW_LEFT_KEY, ARROW_RIGHT_KEY].includes(event.key))) {
+      return
+    }
+
+    const children = this._getChildren()
+
+    const nextActiveElement = getNextActiveElement(children, event.target, event.key === ARROW_RIGHT_KEY, true)
+    Tab.getOrCreateInstance(nextActiveElement).show()
+  }
+
+  _getChildren() { // collection of inner elements
+    return SelectorEngine.find(SELECTOR_INNER_ELEM, this._parent)
+  }
+
+  _getActiveElem() {
+    return this._getChildren().find(child => this._elemIsActive(child)) || null
+  }
+
+  _setInitialAttributes(parent, children) {
+    this._setAttributeIfNotExists(parent, 'role', 'tablist')
+
+    for (const child of children) {
+      this._setInitialAttributesOnChild(child)
     }
+  }
 
-    reflow(element)
+  _setInitialAttributesOnChild(child) {
+    child = this._getInnerElement(child)
+    const isActive = this._elemIsActive(child)
+    const outerElem = this._getOuterElement(child)
+    child.setAttribute('aria-selected', isActive)
 
-    if (element.classList.contains(CLASS_NAME_FADE)) {
-      element.classList.add(CLASS_NAME_SHOW)
+    if (outerElem !== child) {
+      this._setAttributeIfNotExists(outerElem, 'role', 'presentation')
     }
 
-    let parent = element.parentNode
-    if (parent && parent.nodeName === 'LI') {
-      parent = parent.parentNode
+    if (!isActive) {
+      child.setAttribute('tabindex', '-1')
     }
 
-    if (parent && parent.classList.contains(CLASS_NAME_DROPDOWN_MENU)) {
-      const dropdownElement = element.closest(SELECTOR_DROPDOWN)
+    this._setAttributeIfNotExists(child, 'role', 'tab')
 
-      if (dropdownElement) {
-        for (const dropdown of SelectorEngine.find(SELECTOR_DROPDOWN_TOGGLE, dropdownElement)) {
-          dropdown.classList.add(CLASS_NAME_ACTIVE)
-        }
-      }
+    // set attributes to the related panel too
+    this._setInitialAttributesOnTargetPanel(child)
+  }
+
+  _setInitialAttributesOnTargetPanel(child) {
+    const target = getElementFromSelector(child)
 
-      element.setAttribute('aria-expanded', true)
+    if (!target) {
+      return
     }
 
-    if (callback) {
-      callback()
+    this._setAttributeIfNotExists(target, 'tabindex', '0')
+    this._setAttributeIfNotExists(target, 'role', 'tabpanel')
```

Comment:
I would actually leave this aspect out, leaving it up to the users to set their tabpanels to be focusable in their own markup. reason being that this *can* be quite surprising/counterproductive, particularly if the tabpanel itself already has a focusable element right at the start - see the `Tab` behaviour https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-21 second bullet

> When the tab list contains the focus, moves focus to the next element in the page tab sequence outside the tablist, which is the tabpanel *unless the first element containing meaningful content inside the tabpanel is focusable.*

We can add something in the documentation text itself strongly advising authors to make them focusable unless they have a reason not to - that gives them some options

Reply:
just the `tabindex` one (sorry, i see i missed the correct line by one when adding my comment). leave the `role` and `aria-labelledby` ones there